.github/labeler.yml

```diff
@@ -0,0 +1,144 @@
+# list of constraints / globs and then matches all of the constraints for either `any` of the files or
+# `all` of the files.
+#
+# However, `any`/`all` are not supported in a released version and testing off of the `main` branch
+# had them not working proprly. While we wait for this issue to be handled upstream, we can remove
+# the negative / not matches for now and at least have labels again.
+INFRA:
+  - ".github/**/*"
+  - "appveyor.yml"
+  - "tools/**/*"
+  - "dev/create-release/**/*"
+  - ".asf.yaml"
+  - ".gitattributes"
+  - ".gitignore"
+  - "dev/github_jira_sync.py"
+  - "dev/merge_spark_pr.py"
+  - "dev/run-tests-jenkins*"
+BUILD:
+ # Can be supported when a stable release with correct all/any is released
+ #- any: ['dev/**/*', '!dev/github_jira_sync.py', '!dev/merge_spark_pr.py', '!dev/.rat-excludes']
+ - "dev/**/*"
+ - "build/**/*"
+ - "project/**/*"
+ - "assembly/**/*"
+ - "**/*pom.xml"
+ - "bin/docker-image-tool.sh"
+ - "bin/find-spark-home*"
+ - "scalastyle-config.xml"
+ # These can be added in the above `any` clause (and the /dev/**/* glob removed) when
+ # `any`/`all` support is released
+ # - "!dev/github_jira_sync.py"
+ # - "!dev/merge_spark_pr.py"
+ # - "!dev/run-tests-jenkins*"
+ # - "!dev/.rat-excludes"
+DOCS:
+  - "docs/**/*"
+  - "**/README.md"
+  - "**/CONTRIBUTING.md"
+EXAMPLES:
+  - "examples/**/*"
+  - "bin/run-example*"
+# CORE needs to be updated when all/any are released upstream.
+CORE:
+  # - any: ["core/**/*", "!**/*UI.scala", "!**/ui/**/*"] # If any file matches all of the globs defined in the list started by `any`, label is applied.
+  - "core/**/*"
+  - "common/kvstore/**/*"
+  - "common/network-common/**/*"
+  - "common/network-shuffle/**/*"
+  - "python/pyspark/**/*.py"
+  - "python/pyspark/tests/**/*.py"
+SPARK SUBMIT:
+  - "bin/spark-submit*"
+SPARK SHELL:
+  - "repl/**/*"
+  - "bin/spark-shell*"
+SQL:
+#- any: ["**/sql/**/*", "!python/pyspark/sql/avro/**/*", "!python/pyspark/sql/streaming.py", "!python/pyspark/sql/tests/test_streaming.py"]
+  - "**/sql/**/*"
+  - "common/unsafe/**/*"
+  #- "!python/pyspark/sql/avro/**/*"
+  #- "!python/pyspark/sql/streaming.py"
+  #- "!python/pyspark/sql/tests/test_streaming.py"
+  - "bin/spark-sql*"
+  - "bin/beeline*"
+  - "sbin/*thriftserver*.sh"
+  - "**/*SQL*.R"
+  - "**/DataFrame.R"
+  - "**/*WindowSpec.R"
+  - "**/*catalog.R"
+  - "**/*column.R"
+  - "**/*functions.R"
+  - "**/*group.R"
+  - "**/*schema.R"
+  - "**/*types.R"
+AVRO:
+  - "external/avro/**/*"
+  - "python/pyspark/sql/avro/**/*"
+DSTREAM:
+  - "streaming/**/*"
+  - "data/streaming/**/*"
+  - "external/kinesis*"
+  - "external/kafka*"
+  - "python/pyspark/streaming/**/*"
+GRAPHX:
+  - "graphx/**/*"
+  - "data/graphx/**/*"
+ML:
+  - "**/ml/**/*"
+  - "**/*mllib_*.R"
+MLLIB:
+  - "**/spark/mllib/**/*"
+  - "mllib-local/**/*"
+  - "python/pyspark/mllib/**/*"
+STRUCTURED STREAMING:
+  - "sql/**/streaming/**/*"  # TODO - Does this one need a leading **/ or */ ?
```

Comment:
To test for possible matches, I've been using `find` and `grep`. For this one, I believe we likely do need an initial glob as I found the following matches. I will add another test against one of these files.

```bash
$ find . -type d -iname 'streaming' | grep sql
./python/test_support/sql/streaming
./examples/src/main/python/sql/streaming
./examples/src/main/java/org/apache/spark/examples/sql/streaming
./examples/src/main/scala/org/apache/spark/examples/sql/streaming
./sql/core/src/test/java/test/org/apache/spark/sql/streaming
./sql/core/src/test/scala/org/apache/spark/sql/streaming
./sql/core/src/test/scala/org/apache/spark/sql/execution/streaming
./sql/core/src/main/java/org/apache/spark/sql/streaming
./sql/core/src/main/java/org/apache/spark/sql/execution/streaming
./sql/core/src/main/scala/org/apache/spark/sql/streaming
./sql/core/src/main/scala/org/apache/spark/sql/execution/streaming
./sql/core/src/main/scala-2.13/org/apache/spark/sql/execution/streaming
./sql/core/src/main/scala-2.12/org/apache/spark/sql/execution/streaming
./sql/catalyst/src/test/java/org/apache/spark/sql/streaming
./sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/streaming
./sql/catalyst/src/main/java/org/apache/spark/sql/streaming
./sql/catalyst/src/main/java/org/apache/spark/sql/connector/write/streaming
./sql/catalyst/src/main/java/org/apache/spark/sql/connector/read/streaming
./sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/streaming
```

Reply:
As demonstrated in this PR, a leading `**` is needed to match the entries that are not in the root sql directory. https://github.com/kbendick/spark/pull/23

Reply:
I'm pushing a change for this as I feel that changes to `./examples/src/main/scala/org/apache/spark/examples/sql/streaming/...` should be labeled as STRUCTURED STREAMING. It also matches the original configuration better in my opnion.

Reply:
This has been updated to `"**/sql/**/streaming/**/*"` and the TODO comment has been removed.

Reply:
SGTM